File: 704-binary-search/704-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int left = 0, right = nums.size()-1, mid = (left+right)/2;
        while(left < right) {
            if(nums[mid] < target) {
                left = mid+1;
            } else { // nums[mid] >= target
                right = mid;
            }
            mid = (left+right)/2;
        }
        return nums[mid] == target? mid : -1;
    }
};

File: 704-binary-search/704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left, right = 0, len(nums)
        mid = (left+right)//2
        
        while left < right:
            if nums[mid] < target:
                left = mid+1
            elif nums[mid] > target:
                right = mid
            else:
                break

            mid = (left+right)//2
        
        return mid if mid in range(len(nums)) and nums[mid] == target else -1